src/main/java/org/springframework/web/servlet/DispatcherServlet.java

```diff
@@ -0,0 +1,1493 @@
+ *
+ * <li>If a {@link View} or view name is not supplied by the user, then the configured
+ * {@link RequestToViewNameTranslator} will translate the current request into a view name.
+ * The corresponding bean name is "viewNameTranslator"; the default is
+ * {@link org.springframework.web.servlet.view.DefaultRequestToViewNameTranslator}.
+ *
+ * <li>The dispatcher's strategy for resolving multipart requests is determined by a
+ * {@link org.springframework.web.multipart.MultipartResolver} implementation.
+ * Implementations for Apache Commons FileUpload and Servlet 3 are included; the typical
+ * choice is {@link org.springframework.web.multipart.commons.CommonsMultipartResolver}.
+ * The MultipartResolver bean name is "multipartResolver"; default is none.
+ *
+ * <li>Its locale resolution strategy is determined by a {@link LocaleResolver}.
+ * Out-of-the-box implementations work via HTTP accept header, cookie, or session.
+ * The LocaleResolver bean name is "localeResolver"; default is
+ * {@link org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver}.
+ *
+ * <li>Its theme resolution strategy is determined by a {@link ThemeResolver}.
+ * Implementations for a fixed theme and for cookie and session storage are included.
+ * The ThemeResolver bean name is "themeResolver"; default is
+ * {@link org.springframework.web.servlet.theme.FixedThemeResolver}.
+ * </ul>
+ *
+ * <p><b>NOTE: The {@code @RequestMapping} annotation will only be processed if a
+ * corresponding {@code HandlerMapping} (for type-level annotations) and/or
+ * {@code HandlerAdapter} (for method-level annotations) is present in the dispatcher.</b>
+ * This is the case by default. However, if you are defining custom {@code HandlerMappings}
+ * or {@code HandlerAdapters}, then you need to make sure that a corresponding custom
+ * {@code RequestMappingHandlerMapping} and/or {@code RequestMappingHandlerAdapter}
+ * is defined as well - provided that you intend to use {@code @RequestMapping}.
+ *
+ * <p><b>A web application can define any number of DispatcherServlets.</b>
+ * Each servlet will operate in its own namespace, loading its own application context
+ * with mappings, handlers, etc. Only the root application context as loaded by
+ * {@link org.springframework.web.context.ContextLoaderListener}, if any, will be shared.
+ *
+ * <p>As of Spring 3.1, {@code DispatcherServlet} may now be injected with a web
+ * application context, rather than creating its own internally. This is useful in Servlet
+ * 3.0+ environments, which support programmatic registration of servlet instances.
+ * See the {@link #DispatcherServlet(WebApplicationContext)} javadoc for details.
+ *
+ * <p>OSF CAS Customization: introduce a new private field {@link OsfUrlProperties osfUrlProperties} to class
+ * {@link DispatcherServlet}; this allows method {@link this#resolveViewName(String, Map, Locale, HttpServletRequest)}
+ * to send OSF URLs down to the view via model if resolved successfully.</p>
```

Comment:
For CR: again, explains why `osf-cas` customize `DispatcherServlet`, changes in this class is the "real" code change while the previous two are for configuration purpose only.